pkg/install/check/check.go

```diff
@@ -0,0 +1,392 @@
+package check
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/acorn-io/acorn/pkg/k8sclient"
+	"github.com/acorn-io/acorn/pkg/system"
+	"github.com/acorn-io/baaah/pkg/restconfig"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CheckResult describes the results of a check, making it human-readable
+type CheckResult struct {
+	Message string `json:"message"`
+	Passed  bool   `json:"passed"`
+	Name    string `json:"name"`
+}
+
+// PreflightChecks is a list of all checks that are run before the installation.
+// They are crictial and will make the installation fail.
+func PreflightChecks() []CheckResult {
+	return RunChecks(CheckRBAC, CheckNodesReady)
+}
+
+// InFlightChecks is a list of all checks that are run after the installation.
+// They are not critical and should not affect the installation process.
+func InFlightChecks() []CheckResult {
+	checks := []func() CheckResult{
+		CheckDefaultStorageClass,
+		CheckIngressCapability,
+	}
+
+	// Some debugging test
+	if os.Getenv("ACORN_INSTALL_FAIL_CHECKS") == "true" {
+		checks = append(checks, func() CheckResult {
+			return CheckResult{Name: "FailTest", Passed: false, Message: "This is a test failure"}
+		})
+	}
+
+	return RunChecks(checks...)
+}
+
+// IsFailed is a simple helper function marking a list of check results
+// as failed if one or more results show failed status.
+func IsFailed(results []CheckResult) bool {
+	for _, r := range results {
+		if !r.Passed {
+			return true
+		}
+	}
+	return false
+}
+
+// RunChecks runs a list of checks and returns their results as a list.
+func RunChecks(checks ...func() CheckResult) []CheckResult {
+	var results []CheckResult
+	for _, check := range checks {
+		results = append(results, check())
+	}
+	return results
+}
+
+// silenceKlog is a helper function to silence klog output which could disturb
+// the clean installation logs.
+func silenceKlog() {
+	klog.SetOutput(io.Discard)
+	klogv2.SetOutput(io.Discard)
+	utilruntime.ErrorHandlers = nil
+}
+
+// newClient is a helper function to quickly create a new k8s client instance
+func newClient() (client.WithWatch, error) {
```

Comment:
this function seems to be basically the same as k8sclient.Default()

Reply:
replaced :heavy_check_mark: